package/MDAnalysis/analysis/hydrogenbonds/hbond_analysis.py

```diff
@@ -536,11 +536,11 @@ def count_by_type(self):
         acceptor atoms in a hydrogen bond.
         """
 
-        d = self.u.atoms[self.hbonds[:, 1].astype(np.int)]
-        a = self.u.atoms[self.hbonds[:, 3].astype(np.int)]
+        d = self.u.atoms[self.hbonds[:, 1].astype(int)]
```

Comment:
Looking at these (and in general when an integer is made for indexing something), I think we want `np.intp` here

Reply:
done (I think this was the only case it's a thing) - note it does mean that the change here diverges from `develop`, do we want to "reverse backport" it?